packages/compass/src/app/index.tsx

```diff
@@ -91,6 +94,59 @@ import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
 import { getAppName, getAppVersion } from '@mongodb-js/compass-utils';
 const { log, mongoLogId, track } = createLoggerAndTelemetry('COMPASS-APP');
 
+const WithPreferencesAndLoggerProviders: React.FC = ({ children }) => {
+  const loggerProviderValue = {
+    createLogger: createLoggerAndTelemetry,
+    preferences: defaultPreferencesInstance,
+  };
+  return (
+    <PreferencesProvider value={loggerProviderValue.preferences}>
+      <LoggerAndTelemetryProvider value={loggerProviderValue}>
+        {children}
+      </LoggerAndTelemetryProvider>
+    </PreferencesProvider>
+  );
+};
+
+const WithAtlasProviders: React.FC = ({ children }) => {
+  return (
+    <AtlasAuthServiceProvider value={new CompassAtlasAuthService()}>
```

Comment:
We kinda know this never re-renders after initial render here, but just to be absolutely sure our component never creates a new instance after mounting

```suggestion
  const authService = useRef(new CompassAtlasAuthService())
  return (
    <AtlasAuthServiceProvider value={authService.current}>
```

Reply:
added in [cde31be](https://github.com/mongodb-js/compass/pull/5512/commits/cde31be18d2876ae69685be93d307a825b171053)